libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -56,11 +56,10 @@ class IRGenerationContext
 	):
 		m_evmVersion(_evmVersion),
 		m_revertStrings(_revertStrings),
-		m_optimiserSettings(std::move(_optimiserSettings)),
-		m_functions(std::make_shared<MultiUseYulFunctionCollector>())
+		m_optimiserSettings(std::move(_optimiserSettings))
 	{}
 
-	std::shared_ptr<MultiUseYulFunctionCollector> functionCollector() const { return m_functions; }
+	MultiUseYulFunctionCollector& functionCollector() { return m_functions; }
```

Comment:
Why did you also drop off the `const` modifier here? If you need a non-const version, I'd instead add a non-const version but not replace it? hmm...

Reply:
And even if you only want to get the collected functions, the function collector will automatically reset itself - again non-const. So no use for a const version of this.